lib/saml_idp/xml_security.rb

```diff
@@ -100,17 +114,25 @@ def find_base64_cert(options)
         if cert_element
           return cert_element.text unless cert_element.text.blank?
 
-          raise ValidationError.new("Certificate element present in response (ds:X509Certificate) but evaluating to nil")
+          raise ValidationError.new(
+            "Certificate element present in response (ds:X509Certificate) but evaluating to nil",
+            :present_but_nil
+          )
         elsif options[:cert]
           if options[:cert].is_a?(String)
             options[:cert]
           elsif options[:cert].is_a?(OpenSSL::X509::Certificate)
             Base64.encode64(options[:cert].to_pem)
           else
-            raise ValidationError.new("options[:cert] must be Base64-encoded String or OpenSSL::X509::Certificate")
+            raise ValidationError.new(
+              'options[:cert] must be Base64-encoded String or OpenSSL::X509::Certificate',
+              :not_base64_or_cert
+            )
           end
         else
-          raise ValidationError.new("Certificate element missing in response (ds:X509Certificate) and not provided in options[:cert]")
+          raise ValidationError.new(
+            'Certificate element missing in response (ds:X509Certificate) and not provided in options[:cert]',
+            :cert_missing)
```

Comment:
for consistency with other spots do we want to bump this paren to the next line?
```suggestion
            :cert_missing
          )
```

Reply:
fixed in https://github.com/18F/saml_idp/pull/86/commits/c27da1cc3e0e280a8430980d92e290440a220ef5